reference/6/Microsoft.PowerShell.Core/About/about_PowerShell_Config.md

```diff
@@ -0,0 +1,350 @@
+---
+ms.date:  11/02/2018
+schema:  2.0.0
+locale:  en-us
+keywords:  powershell
+title:  about_PowerShell_Config
+---
+
+# About PowerShell Config
+
+## SHORT DESCRIPTION
+Cross-platform configuration files for PowerShell Core.
+
+## LONG DESCRIPTION
+
+The `powershell.config.json` file contains configuration settings for
+PowerShell Core. PowerShell loads this configuration at startup. The settings
+can also be modified at runtime. Previously, these settings were stored in the
+Windows Registry for Windows PowerShell.
+
+> [!WARNING]
+> If the `powershell.config.json` file contains invalid JSON,
+> unsupported keys, or values, PowerShell cannot start an interactive session.
+> If this occurs, you must fix the configuration file.
+
+### System-wide configuration
+
+A `powershell.config.json` file in the `$PSHOME` directory defines the
+configuration for all PowerShell Core sessions running from that PowerShell
+Core installation.
+
+> [!NOTE]
+> The system `powershell.config.json` location is defined as
+> the same directory as the executing System.Management.Automation.dll
+> assembly. This applies to hosted PowerShell SDK instances as well.
+
+### User configurations
+
+You can also configure PowerShell on a per-user basis by placing the file in
+the user-scope configuration directory. The user configuration directory can
+be found across platforms with the command
+`Split-Path $PROFILE.CurrentUserCurrentHost`.
+
+## General configuration settings
+
+### ExecutionPolicy
+
+Configures the execution policy for PowerShell sessions, determining what
+scripts can be run. By default, PowerShell uses the existing execution policy.
+
+> [!NOTE]
+> The [`Set-ExecutionPolicy`](../../Microsoft.PowerShell.Security/Set-ExecutionPolicy.md)
+> cmdlet modifies this setting in the configuration file.
+
+```Schema
+"<shell-id>:ExecutionPolicy": "<execution-policy>"
+```
+
+Where:
+
+- `<shell-id>` refers to the ID of the current PowerShell host.
+  For normal PowerShell Core, this is `Microsoft.PowerShell`.
+  In any PowerShell session, you can discover it with `$ShellId`.
+- `<execution-policy>` refers to a valid execution policy name.
+
+The following example sets the execution policy of PowerShell to `RemoteSigned`.
+
+```json
+{
+  "Microsoft.PowerShell.ExecutionPolicy": "RemoteSigned"
+}
+```
+
+In Windows, the equivalent registry keys can be found in
+`\SOFTWARE\Microsoft\PowerShell\1\ShellIds\Microsoft.PowerShell` under
+`HKEY_LOCAL_MACHINE` and `HKEY_CURRENT_USER`.
+
+### ModulePath
+
+Sets the **PSModulePath** environment variable when PowerShell starts up.
+The default value is `$env:ProgramFiles/PowerShell/Modules`.
```

Comment:
It's still not clear to me. Will `$env:PSModulePath` be the same as the value from the `PSModulePath` property? If so, the default value shouldn't be `$env:ProgramFiles/PowerShell/Modules`; if not, then we cannot say it "sets PSModulePath environment variable".

Reply:
Agreed this isn't clear at all.

The actual code is [here](https://github.com/PowerShell/PowerShell/blob/52aef57ab682b72cea4a0168d7cf20ba554bf6ce/src/System.Management.Automation/engine/PSConfiguration.cs#L98-L113).

It does this:
1. Tries to read the value with key "PSModulePath" from the config.json
1. Has default value of `null`
1. If the result is not `null`, runs `System.Environment.ExpandEnvironmentVariables()` on the value from the JSON file. (Which has the glaring problem to me that there's no way to embed the current $PSHome in it).

With all this, I propose:

> Overrides a PSModulePath component for this PowerShell session. If the configuration is for the current user, sets the CurrentUser module path. If the configuration is for all users, sets the AllUser module path.
>
> If no value is set, the default value for the respective module path component will be used. See [about_Modules](./about_Modules.md#module-and-dsc-resource-locations-and-psmodulepath) for more details on these defaults.
>
> This setting allows environment variables to be used by embedding them between `%` characters, like `"%HOME%\Documents\PowerShell\Modules"`, in the same way as CMD allows. This syntax also applies on Linux and macOS. See below for examples.

Reply:
Reading through the module path link I have there, it looks like we should update that too. But I'm assuming for now it's the best doc link we've got.